mitodl/edx-sysadmin
# switching from mongo to mysql and adding threshold to number of stored git-logs

#### Pre-Flight checklist

#### What are the relevant tickets?
https://github.com/mitodl/edx-sysadmin/issues/12

#### What's this PR do?
* Switches from Mongo to MySQL for GitLogs
* Add threshold (SYSADMIN_MAX_GIT_LOGS_THRESHOLD) value to remove old GitLogs per course

#### How should this be manually tested?
* Test the whole git import and git log feature to test if MySQL works perfectly with GitLogs.
* Set `SYSADMIN_MAX_GIT_LOGS_THRESHOLD` variable to a specific `integer` through your `lms.yml` or `private.py` in `edx-platform` and then import a course. It should delete the old course logs of that specific course if there are more than the defined threshold.

#### Any background context you want to provide?
The main issue we had with MySQL earlier was that the size of Course-Import Logs could get longer than the allowed TextField size (https://electrictoolbox.com/maximum-length-mysql-text-field-types/) so we had to use MongoDB for storing logs.

Now as we have JSONField in MySQL we can use that to solve the problem and get rid of MongoDB. The size of JSONField in MySQL is directly dependent on `max-allowed-packet` global variable (https://stackoverflow.com/questions/40711101/max-json-column-length-in-mysql) which can range up to 1GB and can be easily changed (https://docs.oracle.com/cd/E19182-01/820-6323/gicxk/index.html). So this PR uses that MySQL JSONField.
